src/game_logic/player-actions.ts

```diff
@@ -14,8 +14,15 @@ export function playerInitiateProject (state: GameState, tileIndex: number, proj
     }
   })
 
+  const workers = state.player.resources.workers
+
+  const newWorkers = produce(workers, draft => {
+    draft.free = draft.free - 1
+  })
+
   return produce(state, draft => {
     draft.map.tiles[tileIndex] = newTile
+    draft.player.resources.workers = newWorkers
```

Comment:
looking at this code, I realize that I did a poor job earlier of writing `produce` calls. I think the three `produce` calls in this function, for example, can be combined into one I am pretty sure

if we have time, I might make a PR to run through and simplify all the `produce` calls in this file